### PR TITLE
Added Doxygen For PKCS #11.

### PR DESCRIPTION
**Pull request by lundinc2**
This commit syncs the Doxygen code that was used to generate the FreeRTOS.org PKCS #11 Doxygen.

Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: doc/config/pkcs11 [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        doc/lib \
+        libraries/3rdparty/pkcs11 \
+        libraries/abstractions/pkcs11 \
+        libraries/freertos_plus/standard/pkcs11/include \
+        libraries/freertos_plus/standard/pkcs11/src \
+        libraries/abstractions/pkcs11/mbedtls \
+        libraries/abstractions/pkcs11/include \
+        libraries/freertos_plus/standard/utils/include/ \
+        libraries/freertos_plus/standard/utils/src/ \
+        vendors/pc/boards/windows/ports/pkcs11
```
Comment (lundinc2) [inline doc/config/pkcs11]: Windows port is used here, this file will vary between ports but should be okay to use (and has historically) for the purposes of generating the dox

File: doc/config/pkcs11 [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+INPUT = doc \
+        doc/lib \
+        libraries/3rdparty/pkcs11 \
+        libraries/abstractions/pkcs11 \
+        libraries/freertos_plus/standard/pkcs11/include \
+        libraries/freertos_plus/standard/pkcs11/src \
+        libraries/abstractions/pkcs11/mbedtls \
+        libraries/abstractions/pkcs11/include \
+        libraries/freertos_plus/standard/utils/include/ \
+        libraries/freertos_plus/standard/utils/src/ \
```
Comment (qiutongs) [inline doc/config/pkcs11]: A little abnormal to me when the source folder has to be included.

Comment (lundinc2) [reply doc/config/pkcs11]: I will create a ticket to track migrating the doxygen stuff to only header files.

**Pull request merged**